specification/attestation.adoc

```diff
@@ -1,309 +1,886 @@
+provisioned and platform manufacturer endorsed UDS.
+The CDI for the current TCB layer is a cryptographic derivation of a combination
+of the Attesting Environment CDI (i.e. the loading TCB layer) and the current
+TCB layer measurements:
+
+:stem: asciimath
+[.center]
+[stem]
+++++
+CDI_{0} = KDF(UDS_{Len},\ UDS\ ||\ H_{alg}(Meas(TCB_{0}))
+++++
+:stem: asciimath
+[.center]
+[stem]
+++++
+CDI_{N} = KDF(CDI_{Len},\ CDI_{N-1}\ ||\ H_{alg}(Meas(TCB_{N}))
+++++
+
+Asymmetric key pairs can be derived from a CDI in order to generate the
+attestation key for a DICE layer. This key can then be used to attest of the
+next TCB layer.
+
+:stem: asciimath
+[.center]
+[stem]
+++++
+(UDS\_PrivKey,\ UDS\_PubKey) = AsymKDF(UDS)
+++++
+:stem: asciimath
+[.center]
+[stem]
+++++
+(CDI\_PrivKey_{N},\ CDI\_PubKey_{N}) = AsymKDF(CDI_{N})
+++++
+
+A fixed length TCB layer identifier called `CDI_ID` can be derived from its
+attestation key:
+
+:stem: asciimath
+[.center]
+[stem]
+++++
+UDS\_ID = KeyDerivationFunction(ID_{Len}, UDS\_PubKey)
+++++
+:stem: asciimath
+[.center]
+[stem]
+++++
+CDI\_ID_{N} = KeyDerivationFunction(ID_{Len}, CDI\_PubKey_{N})
+++++
+
+When loading the next TCB layer, the attesting layer provides it with both its
+CDI and its attestation certificate.
+
+The TVM guest can obtain an Attestation Evidence from the TSM by calling the
+TG-ABI `sbi_tee_guest_get_evidence()` FID. The guest provides a challenge value
+to the TSM through this call, and that value must be included in the generated
+Evidence. This value allows relying parties to establish the Attestation
+Evidence freshness.
+
+The Evidence is signed by the TSM attestation key, which is HW RoT-rooted.
+A Relying Party can then verify and authenticate the Evidence with platform
+owner or manufacturer provided Endorsements, like e.g. a Trust Anchor.
+
+The TCB extension and evidence collection for a TVM attestation is shown
+below:
 
-* Create TVM attestation structure in a temporary buffer in per-hart 
-confidential memory
-* Populate TVM TcbEvidenceInfo per the TVM measurements recorded by the TSM
-* Copy additional data from CSR <TBD>
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "Layered Attestation architecture for TVMs"]
+image::layered_attestation.svg[]
+
+It is expected that an implementation will provide implementation-specific
+intrinsics to record measurements of the TSM into the firmware RoT for
+measurement to support the AP-TEE layered RTMs attestation of AP-TEE workloads.
+
+==== Evidence Format
+
+The AP-TEE Attestation Evidence uses the IETF Entity Attestation Token
+(<<EAT>>), formatted as an untagged Concise Binary Object Representation
+(<<CBOR>>) Web Token (<<CWT>>).  An AP-TEE EAT profile is proposed to narrow
+the EAT specification for the AP-TEE use case to enable interoperability.
+
+The CWT is composed of one EAT submodule Claims-Set map where the map values
+are nested tokens for the TVM, TSM and Platform Claims.
```

Comment:
I'd remove _"nested"_ as it may create some confusion: the three blobs are in fact well separated into their respective slots :-)

Reply:
Done, thanks.